eng/pipelines/templates/steps/run-and-validate-linting.yml

```diff
@@ -42,12 +28,11 @@ steps:
     inputs:
       mavenPomFile: pom.xml
       ${{ if ne(parameters.SDKType, 'client') }}:
-        options: '$(DefaultOptions) --no-transfer-progress -T 1C -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -pl $(ProjectList)'
-        goals: 'checkstyle:check spotbugs:check'
+        options: '$(DefaultOptions) --no-transfer-progress -T ${{parameters.BuildParallelization}} -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dspotless.apply.skip=true -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.skip=true -pl $(ProjectList) -am'
       ${{ else }}:
-        options: '$(DefaultOptions) --no-transfer-progress -T 1C -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl $(ProjectList)'
-        goals: 'checkstyle:check spotbugs:check revapi:check'
-      mavenOptions: '$(MemoryOptions)'
+        options: '$(DefaultOptions) --no-transfer-progress -T ${{parameters.BuildParallelization}} -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dspotless.apply.skip=true -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl $(ProjectList) -am'
```

Comment:
Should we skip javadoc?

Reply:
For this part we skip Javadoc, there is another Maven task run to run Javadoc later on. This is done as CheckStyle, RevApi, and Spotbugs all have the ability to output a file that contains the issues found but Javadoc only logs to console. So, this one is run in parallel as we don't have a concern for interlaced output then a single thread job is run for Javadoc to ensure the output can be read.